nbextensions/appCell2/widgets/appCellWidget-fsm.js

```diff
@@ -1,6 +1,137 @@
 define([], () => {
     'use strict';
 
+    const uiTabs = {
```

Comment:
Did quite a lot of refactoring on this file and removed a whole load of cruft. Also combined some unnecessary FSM states to simplify things.